### DISCUSSION TITLE
No support for 4k 60fps LTC

### DISCUSSION THREAD

**Discussion by burnbrighter** [gopro/labs#337] [Feature Requests]
My workflow requires LTC for 4k 60fps.  I am using the Tentacle Sync Mk II and am trying to have continual timesync over the audio channel, but am constantly running in to drift.  60fps should be supported as 29.97fps on the GoPro, but its not.  I don't care if it could work via the QR code method or LTC jam sync attached via audio cable: I just need it to work.  Could this PLEASE be added or fixed in the next Hero 10 firmware release?  Thanks

**Comment by dnewman-gpsw** [comment]
Hmm...  LTC does not exist at 60fps.  Tentacle Sync Mk II : "Supports 23.98, 24, 25, 29.97, 29.97DF, and 30 fps frame rates"

However, LTC can be jam sync'd today with the camera in 60fps, with 30fps timecode. That are some conversions that have to happen as 11:44:25:21 (30fps TC) is 11:44:25:42 in 60fps timecode, representing the same moment in time.  GoPro use timecode at native FPS, so the conversion is happening correctly and automatically.  Timecode is a standard GoPro feature, Labs adds reading LTC.

If you want to access the original 29.97 LTC timecode, that is stored in the audio track, for you to use however you need.

So there isn't anything that should need to be fixed.

**Reply by burnbrighter** [reply]
So if I am setting TS to audio LTC at 29.97 fps, but shooting at 60 fps, there will always be a gap of 21 frames?  And how do I account for this over longer time periods?  It sounds like I need to calculate this out in each clip?  Davinci Resolve does have the options for "Update Timecode from Audio Track" option, but I believe the clip will need the offset option.  My gut tells me this is not a linear difference though when thinking in terms of drift.  How do I account for drift later in the clip?  From experience, the further you go in to timeline. the more the drift is - it increases.

**Reply by dnewman-gpsw** [reply]
This is no gap or drift as 1:44:25:21 (30fps TC) is the same as 11:44:25:42 (60fps TC).  21 x 2 = 42, the last digit is frames within a second.

**Reply by burnbrighter** [reply]
Thank you for this information.  Are you saying then, that I will need to account for that difference in Davinci Resolve if I am releasing a 4k 60 fps video?  If yes, is offset the best way to handle this?

**Reply by dnewman-gpsw** [reply]
There should be nothing to account for.

**Reply by dnewman-gpsw** [reply]
May you should have started with what you are doing, which types of cameras are you syncing.  Multiple GoPros or a GoPro with another camera type. Are you using MediaMod or the external mic adaptor?  What drift are you seeing?

**Reply by burnbrighter** [reply]
This is my complete set up:  ZCAM E2 S6, GoPro Hero 10, (new) Tentacle Sync Track E.  The ZCAM and GoPro are shooting at 4k 60 fps and both have Tentacle Sync E Mk II attached to their audio in ports.  All tentacle devices are set to 29.97fps.  With the GoPro, I have the latest beta labs firmware installed (v1.50.70).  The digits go green, so I know it is registering the timecode.  I have both Tentacles time devices set to mic level input (to ensure the audio scratch channel is captured).  I am doing all post-edit in Davinci Resolve 18.1.4, build 9.  I use the timecode sync method, but update the timecode to audio channel (as discussed above).  There is always a delay between of X number of frames between the ZCAM and the GoPro  AND the GoPro and the Tentacle Sync Track E (timecoded audio).  It's pretty easy to see in what I am editing (softball games).  For example, when the ball hits the bat, then sound is off by a few frames.  Further to this, this drift gets worse throughout a 7 inning game (typically 1.5 - 1.75 hours).  I know there are lot of things you don't support in my formula above, but I am kind of lost on how I should approach this and try to make this work.  Thanks for any assistance you may provide.

**Reply by dnewman-gpsw** [reply]
Are you starting and stopping capture, or capturing continuously on both or one of the cameras? How much is the drift per hour?  Is it consistent? Timecode in the MP4/MOV, is the timecode for the first frame only. As the two cameras have independent hardware clocks, they will drift compared to each other, this is normal. Cameras don't have atomic clocks, and timecode data is not a clock source for video frame timing.  The crystals used for camera timing might be good for around 20ppm (20 parts per million). So a worst case of +/- 20ppm is +/- 1.7 second per day of clock drift, or 4.3 frames per hour at 60fps. However, if captures are regularly started and stopped (like in a film production) each new capture gets a new timecode for the first frame, and be as precise as the time source (Tentacle Sync in this case.) A long example of a 10 minute reel, would be frame accurate for that period, even with a 20ppm clock difference between cameras. So I think you are seeing camera clock differences, not timecode drift.  

Technically if you use the audio timecode within your editing tool, instead of the MP4 first frame timecode, you might be able to overcome the clock differences. The editor would have to drop or duplicate frames to maintain sync, as technically, relative to each other, one camera is shooting 59.9412 and the other 59.9388 fps. However this is a question for support for your editing tool, as it is not a workflow I have ever tried. If your captures are started and stopped with each batter, your timecode sync should be pretty good. If there a few frame of difference that should be consistent, so the whole camera track can be shifted to fix that.
